status/barcode.py

```diff
@@ -23,45 +23,89 @@ def get(self):
-                        print_barcode(createdLabel_joined)
+                        print_barcode(linesToPrint)
+                else: # file submitted is a text file
+                    array_linesToPrint = linesToPrint.splitlines() # split into the different lines of the text file
+                    for line in array_linesToPrint:
+                        createdLabel = makeBarcode(line, match_barcode(line, False))
+                        createdLabel_joined = '\n'.join(createdLabel)
+                        for _ in range(copies): # loops over copies to print
+                            print_barcode(createdLabel_joined)
+
+            # printing from string input
+            text_for_printing = self.get_argument('text_to_print') # retrieves string from form
+            withBarcode = self.get_argument('print_with_barcode') #retrieves information is the label should be printed with barcodes
+            if(withBarcode == "noBC"):
+                barcode = False
+            else:
+                barcode = True
+            if len(text_for_printing) > 0 : # check wether there was an entry or not
+                createdLabel = makeBarcode(text_for_printing, match_barcode(text_for_printing, barcode))
+                createdLabel_joined = '\n'.join(createdLabel)
+                for _ in range(copies): # loops over copies to print
+                    print_barcode(createdLabel_joined)
+            else:
+                print("no text was given")
+
+            self.set_status(200)
+            self.set_header("Content-type", "application/json")
+            self.finish({'message': 'Submitted to printer!'})
+
+        elif (labelType == "user_labels"):
+            # printing user project Labels
+            user_project_ID = self.get_argument('projectLabel_to_print')
+            startP = self.get_argument('plate_start') # choose the plate number to start, default = 1
+            endP = self.get_argument('plate_end') # choose the plate number to end, default = 5
+            # if labels for more than one project are to be printed, this prints another
+            # set of barcodes by adding one to the previous project number
+            projectNo = self.get_argument('numberOfProjects')
 
-        # printing from string input
-        text_for_printing = self.get_argument('text_to_print') # retrieves string from form
-        if len(text_for_printing) > 0 : # check wether there was an entry or not
-            createdLabel = makeBarcode(text_for_printing, match_barcode(text_for_printing, False))
-            createdLabel_joined = '\n'.join(createdLabel)
-            for _ in range(copies): # loops over copies to print
-                print_barcode(createdLabel_joined)
-        else:
-            print("no text was given")
+            if len(user_project_ID) > 0 : # check that there is an entry
+                isValidUserID = match_userID(user_project_ID) # check that (only) a project ID was given (P12345 instead of P12345P1)
```

Comment:
re.compile(r'P\d+$') will match both P12345 and P12345P1 😅

Reply:
oooooh. I thought the \d only allows digits?

Reply:
yes, it will match P1 at the end though, since there's no requirement for the P to be at the beginning of the string.

Reply:
*facepalm*
of course